viam.errors
===========

.. py:module:: viam.errors


Exceptions
----------

.. autoapisummary::

   viam.errors.ViamError
   viam.errors.InsecureConnectionError
   viam.errors.DuplicateResourceError
   viam.errors.ViamGRPCError
   viam.errors.ResourceNotFoundError
   viam.errors.MethodNotImplementedError
   viam.errors.NotSupportedError
   viam.errors.ValidationError
   viam.errors.NoCaptureToStoreError


Module Contents
---------------

.. py:exception:: ViamError

   Bases: :py:obj:`Exception`


   Base Viam Error


.. py:exception:: InsecureConnectionError(address: str, authenticated: bool = False)

   Bases: :py:obj:`ViamError`


   Exception raised when establishing an insecure connection

   .. attribute:: address

      The endpoint of the connection

   .. attribute:: authenticated

      True if the insecure connection provided credentials

   .. attribute:: message

      Explanation of the error


   .. py:attribute:: address


   .. py:attribute:: authenticated


   .. py:attribute:: message


.. py:exception:: DuplicateResourceError(name: str)

   Bases: :py:obj:`ViamError`


   Exception raised when attempting to register a resource
   with the same name as an existing resource already in
   the registry


   .. py:attribute:: name


   .. py:attribute:: message


.. py:exception:: ViamGRPCError(message: str = '', grpc_code: grpclib.Status = Status.INTERNAL)

   Bases: :py:obj:`ViamError`


   Exception raised if it could happen as a part of GRPC calls.


   .. py:attribute:: message


   .. py:attribute:: grpc_code


   .. py:property:: grpc_error
      :type: grpclib.GRPCError



.. py:exception:: ResourceNotFoundError(resource: str, name: str)

   Bases: :py:obj:`ViamGRPCError`


   Exception raised when a resource is not found in the registry


   .. py:attribute:: resource


   .. py:attribute:: name


   .. py:attribute:: message


   .. py:attribute:: grpc_code


.. py:exception:: MethodNotImplementedError(method_name: str)

   Bases: :py:obj:`ViamGRPCError`


   Exception raised when specific methods are unimplemented.


   .. py:attribute:: message


   .. py:attribute:: grpc_code


.. py:exception:: NotSupportedError(message: str)

   Bases: :py:obj:`ViamGRPCError`


   Exception raised when specific component functionality is not supported


   .. py:attribute:: message


   .. py:attribute:: grpc_code


.. py:exception:: ValidationError(message: str)

   Bases: :py:obj:`ViamGRPCError`


   Exception raised when there is an error during module validation


   .. py:attribute:: message


   .. py:attribute:: grpc_code


.. py:exception:: NoCaptureToStoreError

   Bases: :py:obj:`ViamGRPCError`


   Exception raised in filter module to signal that data capture should not be stored


   .. py:attribute:: message
      :value: 'no capture from filter module'



   .. py:attribute:: grpc_code


